src/app/games/martian-chess/MartianChessRules.ts

```diff
@@ -0,0 +1,192 @@
+import { Coord } from 'src/app/jscaip/Coord';
+import { Direction, Vector } from 'src/app/jscaip/Direction';
+import { MGPNode } from 'src/app/jscaip/MGPNode';
+import { Player } from 'src/app/jscaip/Player';
+import { GameStatus, Rules } from 'src/app/jscaip/Rules';
+import { RulesFailure } from 'src/app/jscaip/RulesFailure';
+import { assert } from 'src/app/utils/assert';
+import { MGPFallible } from 'src/app/utils/MGPFallible';
+import { MGPMap } from 'src/app/utils/MGPMap';
+import { MGPOptional } from 'src/app/utils/MGPOptional';
+import { MartianChessMove, MartianChessMoveFailure } from './MartianChessMove';
+import { MartianChessPiece, MartianChessState } from './MartianChessState';
+
+export class MartianChessRulesFailure {
+
+    public static readonly MUST_CHOOSE_PIECE_FROM_YOUR_TERRITORY: () => string = () => $localize`Choose a piece from your territory to move it`;
+
+    public static readonly CANNOT_CAPTURE_YOUR_OWN_PIECE_NOR_PROMOTE_IT: () => string = () => $localize`This move is not a valid promotion nor a valid capture`;
+
+    public static readonly CANNOT_UNDO_LAST_MOVE: () => string = () => $localize`You cannot do a move that is the reversed of the last one`;
```

Comment:
You cannot *perform* a move that is the *reverse* of the *previous* one
Et point final